### DISCUSSION TITLE
Detect company based on email at registration / auto join?

### DISCUSSION THREAD

**Discussion by RicLeP** [andrewdwallo/filament-companies#100] [Q&A] [answered]
I’m working on a project (rebuilding a legacy application) where different companies can sign up. In the past we’ve had problems where a user has left the company, a new member of staff has come aboard and didn’t know the company had an account. So they have created a new account leading to duplicate companies in the system.

Is there an elegant solution to deal with this? I was thinking using the email domain and checking if there was a user and then company matching this. It’s not ideal as they may have several companies under a single domain and would need to be disabled for gmail addresses etc. I think automatically adding them would be too risky, instead they get offered the company to join. But then they should really be approved by an admin (who may have left the company so can’t).

I hope that makes sense. Do you have any thoughts or ideas around tackling this? I’m not entirely sure where it would live in the code too as new to Filament.

**Comment by andrewdwallo** [comment]
Well so are you not deleting the company if the user has "left"? This way there wouldn't be duplicate companies?

**Reply by RicLeP** [reply]
I’ll not be directly managing it, it’ll be mostly self-service so if an account is created for a company and then the administrator leaves (in real life) and they employ a new person who then wants to join the website they may not have prior knowledge that the former employee created an account so makes a new company. I want to try and smooth the process and provide any messaging to the new user that there is already a company created.

I am thinking I could display a message on the create company screen with the name and email of the administrator for them to contact, or to contact me if they know they no longer work there.

**Reply by andrewdwallo** [reply, marked as answer]
Yeah, I suppose that would be a solid solution. I think this would be up to the company and it is their responsibility to ensure new employees are aware of this. I mean, what is common for popular services to do in this case? Maybe implement something similar.

Just ensure that your approach complies with data privacy regulations and doesn’t expose sensitive information. For instance, when displaying admin contact information, ensure it's done securely and with consent.
